ba/cheatsheets/index.md

```diff
@@ -0,0 +1,86 @@
+language: ba
+---
+
+###### Doprinio {{ page.by }}
+
+|                                                                                                          |                 |
+| ------                                                                                                   | ------          |
+|  <span id="variables" class="h2">varijable</span>                                                                       |                 |
+|  `var x = 5`                                                                                             |  varijabla       |
+|  <span class="label success">Dobro</span> `val x = 5`<br> <span class="label important">Loše</span> `x=6`  |  konstanta       |
+|  `var x: Double = 5`                                                                                     |  eksplicitni tip  |
+|  <span id="functions" class="h2">funkcije</span>                                                                       |                 |
+|  <span class="label success">Dobro</span> `def f(x: Int) = { x*x }`<br> <span class="label important">Loše</span> `def f(x: Int)   { x*x }` |  definicija funkcije <br> skrivena greška: bez = ovo je procedura koja vraća Unit; uzrokuje zabunu |
+|  <span class="label success">Dobro</span> `def f(x: Any) = println(x)`<br> <span class="label important">Loše</span> `def f(x) = println(x)` |  definicija funkcije <br> sintaksna greška: potrebni su tipovi za svaki argument. |
+|  `type R = Double`                                                                                       |  alijas za tip     |
+|  `def f(x: R)` vs.<br> `def f(x: => R)`                                                                  |  poziv-po-vrijednosti <br> poziv-po-imenu (lijeni parameteri) |
+|  `(x:R) => x*x`                                                                                          |  anonimna funkcija  |
+|  `(1 to 5).map(_*2)` vs.<br> `(1 to 5).reduceLeft( _+_ )`                                                |  anonimna funkcija: donja crta odgovara argumentu po poziciji. |
+|  `(1 to 5).map( x => x*x )`                                                                              |  anonimna funkcija: da bi koristili argument više od jednom, morate mu dati ime. |
+|  <span class="label success">Dobro</span> `(1 to 5).map(2*)`<br> <span class="label important">Loše</span> `(1 to 5).map(*2)` |  anonimna funkcija: vezana infiksna metoda. Koristite `2*_` zbog jasnoće. |
+|  `(1 to 5).map { x => val y=x*2; println(y); y }`                                                             |  anonimna funkcija: blokovski stil vraća vrijednost zadnjeg izraza. |
+|  `(1 to 5) filter {_%2 == 0} map {_*2}`                                                                  |  anonimne funkcije: pipeline stil. (može i sa oblim zagradama). |
```

Comment:
Dot (`.`) goes into parenthesis. M is uppercase at the beginning of that sentence.